src/models.rs

```diff
@@ -617,17 +617,22 @@ impl RepositoryUses<'_> {
     ///
     /// This is an asymmetrical match: `actions/checkout@v3` "matches"
     /// the `actions/checkout` template but not vice versa.
+    ///
+    /// Comparisons consider that GitHub does not adopt
+    /// case-sensitive URLs, hence templates like
+    /// `Actions/Checkout` and `actions/checkout`
+    /// resolve to the same Action
     pub(crate) fn matches<'a>(&self, template: impl TryInto<RepositoryUses<'a>>) -> bool {
         let Ok(other) = template.try_into() else {
             return false;
         };
 
-        self.owner == other.owner
-            && self.repo == other.repo
-            && self.subpath == other.subpath
-            && other
-                .git_ref
-                .map_or(true, |git_ref| Some(git_ref) == self.git_ref)
+        self.owner.to_lowercase() == other.owner.to_lowercase()
+            && self.repo.to_lowercase() == other.repo.to_lowercase()
```

Comment:
The `owner` and `repo` could probably be compared with [eq_ignore_ascii_case](https://doc.rust-lang.org/std/primitive.str.html#method.eq_ignore_ascii_case) instead, to avoid a few allocations (since GH requires both to be ASCII).

The subpath and git_ref can be anything OTOH, so those look good to me!

Reply:
Awesome input, justed learned about another small gem in Rust std 🙂

Done [here](https://github.com/woodruffw/zizmor/pull/353/commits/6e76cbfbc2aa2831754b1b9c633ee3de43d1bc7d)